### DISCUSSION TITLE
How to use HTTP extension with Windows ODBC

### DISCUSSION THREAD

**Discussion by djouallah** [duckdb/duckdb#3908] [Q&A]
I have a use case where I want to Query some parquet files using an URL, it works very well in python, but using Excel with ODBC, it does not work as the extension is not installed, and could not find any documentation on how to install it ?

**Comment by Mytherin** [comment]
See [this guide](https://duckdb.org/docs/guides/import/http_import).

**Reply by djouallah** [reply]
@Mytherin it does not work with odbc windows as you can't run multistatement.

**Reply by djouallah** [reply]
![image](https://user-images.githubusercontent.com/12554469/174474844-0a570b28-6579-4edb-9a8b-fa5bb85873a7.png)

**Reply by Mytherin** [reply]
You do not need to run all statements at once. You can run them individually. `INSTALL` should only be run once, and `LOAD` should be run once per open connection. Think of `INSTALL` like `pip install`, and `LOAD` like `import` in Python.

**Comment by djouallah** [comment]
I got this when I try to run install
![image](https://user-images.githubusercontent.com/12554469/174475268-f725f6a1-7ad1-4688-854c-0f58bbed9667.png)

**Reply by Mytherin** [reply]
@tiagokepe could you perhaps have a look at this when you have time? Could we also expand the output here to include the actual error message that the system provides?

**Comment by djouallah** [comment]
updated to the latest version
![image](https://user-images.githubusercontent.com/12554469/174773706-dbf0cd59-cd3f-4d07-9f92-73a58f94aa14.png)

**Comment by djouallah** [comment]
ok it seems to be installed but can't load it
![image](https://user-images.githubusercontent.com/12554469/174775887-958de81e-cf48-45f9-8070-79376f30c9ab.png)

**Reply by Mause** [reply]
What error do you get if you try to load it?

**Comment by djouallah** [comment]
I think Excel expect some data, trying multi statement will not work too as it is not supported
![image](https://user-images.githubusercontent.com/12554469/174789030-879d169c-fd08-43ff-9f4e-30e34f53a64c.png)

**Comment by Alex-Monahan** [comment]
Maybe try using a file based DuckDB instead of an in memory one?

**Comment by tiagokepe** [comment]
Hey @djouallah,

I tested the `httpfs` extension with our ODBC driver on Linux through the ISQL command, it looks like successful as follow:

![image](https://user-images.githubusercontent.com/403295/174860118-44987169-b831-4fe5-a2c1-7e263874f88b.png)

Maybe the issue could be because of Windows, then I grabbed the [duckdb Windows CLI](https://github.com/duckdb/duckdb/releases/download/v0.4.0/duckdb_cli-windows-amd64.zip) from the lastest release, and ran the same commands as follows:

![image](https://user-images.githubusercontent.com/403295/174861932-f7422db3-81fe-420f-b12e-65c34b62a585.png)

Could you test the `INSTALL` and `LOAD` with the duckdb shell as I did in the last print screen?

If successful, could you enable the ODBC trace using the Windows tool: `ODBC Data Source Administrator` (aka `odbcad32.exe`), run Excel again, and atach the ODBC trace file here.
Some instructions for that tool can be found in [this guide](https://duckdb.org/docs/api/odbc/windows#dsn-windows-setup).

**Comment by djouallah** [comment]
it does work with windows CLI, and here is the log for ODBC with PowerQuery, notice that PowerQuery don't like query that does not return results


excel           2f98-2ee4	ENTER SQLAllocHandle 
		SQLSMALLINT                  1 <SQL_HANDLE_ENV>
		SQLHANDLE           0x0000000000000000
		SQLHANDLE *         0x0000021CB40734A8

excel           2f98-2ee4	EXIT  SQLAllocHandle  with return code 0 (SQL_SUCCESS)
		SQLSMALLINT                  1 <SQL_HANDLE_ENV>
		SQLHANDLE           0x0000000000000000
		SQLHANDLE *         0x0000021CB40734A8 ( 0x0000021CAB694B10)

excel           2f98-2ee4	ENTER SQLSetEnvAttr 
		SQLHENV             0x0000021CAB694B10
		SQLINTEGER                 200 <SQL_ATTR_ODBC_VERSION>
		SQLPOINTER                 3 <SQL_OV_ODBC3>
		SQLINTEGER                  -6 

excel           2f98-2ee4	EXIT  SQLSetEnvAttr  with return code 0 (SQL_SUCCESS)
		SQLHENV             0x0000021CAB694B10
		SQLINTEGER                 200 <SQL_ATTR_ODBC_VERSION>
		SQLPOINTER                 3 <SQL_OV_ODBC3>
		SQLINTEGER                  -6 

excel           2f98-2ee4	ENTER SQLSetEnvAttr 
		SQLHENV             0x0000021CAB694B10
		SQLINTEGER                 201 <SQL_ATTR_CONNECTION_POOLING>
		SQLPOINTER                 2 <SQL_CP_ONE_PER_HENV>
		SQLINTEGER                  -6 

excel           2f98-2ee4	EXIT  SQLSetEnvAttr  with return code 0 (SQL_SUCCESS)
		SQLHENV             0x0000021CAB694B10
		SQLINTEGER                 201 <SQL_ATTR_CONNECTION_POOLING>
		SQLPOINTER                 2 <SQL_CP_ONE_PER_HENV>
		SQLINTEGER                  -6 

Microsoft.Mashu 2810-295c	ENTER SQLAllocHandle 
		SQLSMALLINT                  1 <SQL_HANDLE_ENV>
		SQLHANDLE           0x0000000000000000
		SQLHANDLE *         0x00000219552C7CA0

Microsoft.Mashu 2810-295c	EXIT  SQLAllocHandle  with return code 0 (SQL_SUCCESS)
		SQLSMALLINT                  1 <SQL_HANDLE_ENV>
		SQLHANDLE           0x0000000000000000
		SQLHANDLE *         0x00000219552C7CA0 ( 0x0000021952E4A070)

Microsoft.Mashu 2810-295c	ENTER SQLSetEnvAttr 
		SQLHENV             0x0000021952E4A070
		SQLINTEGER                 200 <SQL_ATTR_ODBC_VERSION>
		SQLPOINTER                 3 <SQL_OV_ODBC3>
		SQLINTEGER                  -6 

Microsoft.Mashu 2810-295c	EXIT  SQLSetEnvAttr  with return code 0 (SQL_SUCCESS)
		SQLHENV             0x0000021952E4A070
		SQLINTEGER                 200 <SQL_ATTR_ODBC_VERSION>
		SQLPOINTER                 3 <SQL_OV_ODBC3>
		SQLINTEGER                  -6 

Microsoft.Mashu 2810-295c	ENTER SQLSetEnvAttr 
		SQLHENV             0x0000021952E4A070
		SQLINTEGER                 201 <SQL_ATTR_CONNECTION_POOLING>
		SQLPOINTER                 2 <SQL_CP_ONE_PER_HENV>
		SQLINTEGER                  -6 

Microsoft.Mashu 2810-295c	EXIT  SQLSetEnvAttr  with return code 0 (SQL_SUCCESS)
		SQLHENV             0x0000021952E4A070
		SQLINTEGER                 201 <SQL_ATTR_CONNECTION_POOLING>
		SQLPOINTER                 2 <SQL_CP_ONE_PER_HENV>
		SQLINTEGER                  -6 

Microsoft.Mashu 2810-295c	ENTER SQLAllocHandle 
		SQLSMALLINT                  2 <SQL_HANDLE_DBC>
		SQLHANDLE           0x0000021952E4A070
		SQLHANDLE *         0x0000021955349B58

Microsoft.Mashu 2810-295c	EXIT  SQLAllocHandle  with return code 0 (SQL_SUCCESS)
		SQLSMALLINT                  2 <SQL_HANDLE_DBC>
		SQLHANDLE           0x0000021952E4A070
		SQLHANDLE *         0x0000021955349B58 ( 0x0000021952E4A150)

Microsoft.Mashu 2810-295c	ENTER SQLSetConnectAttrW 
		SQLHDBC             0x0000021952E4A150
		SQLINTEGER                 103 <SQL_ATTR_LOGIN_TIMEOUT>
		SQLPOINTER                15
		SQLINTEGER                  -5 

Microsoft.Mashu 2810-295c	EXIT  SQLSetConnectAttrW  with return code 0 (SQL_SUCCESS)
		SQLHDBC             0x0000021952E4A150
		SQLINTEGER                 103 <SQL_ATTR_LOGIN_TIMEOUT>
		SQLPOINTER                15
		SQLINTEGER                  -5 

Microsoft.Mashu 2810-295c	ENTER SQLDriverConnectW 
		HDBC                0x0000021952E4A150
		HWND                0x0000000000000000
		WCHAR *             0x00007FFA1E8621A0 [      -3] "******\ 0"
		SWORD                       -3 
		WCHAR *             0x00007FFA1E8621A0 
		SWORD                       -3 
		SWORD *             0x0000000000000000
		UWORD                        0 <SQL_DRIVER_NOPROMPT>

Microsoft.Mashu 2810-295c	EXIT  SQLDriverConnectW  with return code 0 (SQL_SUCCESS)
		HDBC                0x0000021952E4A150
		HWND                0x0000000000000000
		WCHAR *             0x00007FFA1E8621A0 [      -3] "******\ 0"
		SWORD                       -3 
		WCHAR *             0x00007FFA1E8621A0 <Invalid buffer length!> [-3]
		SWORD                       -3 
		SWORD *             0x0000000000000000
		UWORD                        0 <SQL_DRIVER_NOPROMPT>

Microsoft.Mashu 2810-295c	ENTER SQLGetInfoW 
		HDBC                0x0000021952E4A150
		UWORD                        6 <SQL_DRIVER_NAME>
		PTR                 0x000002195534A3A0 
		SWORD                      100 
		SWORD *             0x0000007B2CFFDCEC

Microsoft.Mashu 2810-295c	EXIT  SQLGetInfoW  with return code 0 (SQL_SUCCESS)
		HDBC                0x0000021952E4A150
		UWORD                        6 <SQL_DRIVER_NAME>
		PTR                 0x000002195534A3A0 [      12] "DuckDB"
		SWORD                      100 
		SWORD *             0x0000007B2CFFDCEC (12)

Microsoft.Mashu 2810-295c	ENTER SQLGetInfoW 
		HDBC                0x0000021952E4A150
		UWORD                        7 <SQL_DRIVER_VER>
		PTR                 0x000002195534A610 
		SWORD                      100 
		SWORD *             0x0000007B2CFFDCEC

Microsoft.Mashu 2810-295c	EXIT  SQLGetInfoW  with return code 0 (SQL_SUCCESS)
		HDBC                0x0000021952E4A150
		UWORD                        7 <SQL_DRIVER_VER>
		PTR                 0x000002195534A610 [      20] "03.00.0000"
		SWORD                      100 
		SWORD *             0x0000007B2CFFDCEC (20)

Microsoft.Mashu 2810-295c	ENTER SQLGetInfoW 
		HDBC                0x0000021952E4A150
		UWORD                       17 <SQL_DBMS_NAME>
		PTR                 0x000002195534A888 
		SWORD                      100 
		SWORD *             0x0000007B2CFFDCEC

Microsoft.Mashu 2810-295c	EXIT  SQLGetInfoW  with return code 0 (SQL_SUCCESS)
		HDBC                0x0000021952E4A150
		UWORD                       17 <SQL_DBMS_NAME>
		PTR                 0x000002195534A888 [      12] "DuckDB"
		SWORD                      100 
		SWORD *             0x0000007B2CFFDCEC (12)

Microsoft.Mashu 2810-295c	ENTER SQLGetInfoW 
		HDBC                0x0000021952E4A150
		UWORD                       18 <SQL_DBMS_VER>
		PTR                 0x000002195534AAF8 
		SWORD                      100 
		SWORD *             0x0000007B2CFFDCEC

Microsoft.Mashu 2810-295c	EXIT  SQLGetInfoW  with return code 0 (SQL_SUCCESS)
		HDBC                0x0000021952E4A150
		UWORD                       18 <SQL_DBMS_VER>
		PTR                 0x000002195534AAF8 [      24] "v0.4.1-dev33"
		SWORD                      100 
		SWORD *             0x0000007B2CFFDCEC (24)

Microsoft.Mashu 2810-295c	ENTER SQLAllocHandle 
		SQLSMALLINT                  3 <SQL_HANDLE_STMT>
		SQLHANDLE           0x0000021952E4A150
		SQLHANDLE *         0x000002195534D998

Microsoft.Mashu 2810-295c	EXIT  SQLAllocHandle  with return code 0 (SQL_SUCCESS)
		SQLSMALLINT                  3 <SQL_HANDLE_STMT>
		SQLHANDLE           0x0000021952E4A150
		SQLHANDLE *         0x000002195534D998 ( 0x000002196D000530)

Microsoft.Mashu 2810-295c	ENTER SQLGetInfoW 
		HDBC                0x0000021952E4A150
		UWORD                       77 <SQL_DRIVER_ODBC_VER>
		PTR                 0x000002195534D9E8 
		SWORD                      100 
		SWORD *             0x0000007B2CFFDE3C

Microsoft.Mashu 2810-295c	EXIT  SQLGetInfoW  with return code 0 (SQL_SUCCESS)
		HDBC                0x0000021952E4A150
		UWORD                       77 <SQL_DRIVER_ODBC_VER>
		PTR                 0x000002195534D9E8 [      10] "03.00"
		SWORD                      100 
		SWORD *             0x0000007B2CFFDE3C (10)

Microsoft.Mashu 2810-295c	ENTER SQLGetInfoW 
		HDBC                0x0000021952E4A150
		UWORD                        6 <SQL_DRIVER_NAME>
		PTR                 0x000002195534DAE0 
		SWORD                      100 
		SWORD *             0x0000007B2CFFDE7C

Microsoft.Mashu 2810-295c	EXIT  SQLGetInfoW  with return code 0 (SQL_SUCCESS)
		HDBC                0x0000021952E4A150
		UWORD                        6 <SQL_DRIVER_NAME>
		PTR                 0x000002195534DAE0 [      12] "DuckDB"
		SWORD                      100 
		SWORD *             0x0000007B2CFFDE7C (12)

Microsoft.Mashu 2810-295c	ENTER SQLExecDirectW 
		HSTMT               0x000002196D000530
		WCHAR *             0x0000021954D7B08C [      -3] "INSTALL 'httpfs';\ a\ 0"
		SDWORD                    -3

Microsoft.Mashu 2810-295c	EXIT  SQLExecDirectW  with return code 0 (SQL_SUCCESS)
		HSTMT               0x000002196D000530
		WCHAR *             0x0000021954D7B08C [      -3] "INSTALL 'httpfs';\ a\ 0"
		SDWORD                    -3

Microsoft.Mashu 2810-295c	ENTER SQLGetFunctions 
		HDBC                0x0000021952E4A150
		UWORD                     1021 
		UWORD *             0x0000007B2CFFDF40

Microsoft.Mashu 2810-295c	EXIT  SQLGetFunctions  with return code 0 (SQL_SUCCESS)
		HDBC                0x0000021952E4A150
		UWORD                     1021 
		UWORD *             0x0000007B2CFFDF40 (1)

Microsoft.Mashu 2810-295c	ENTER SQLGetInfoW 
		HDBC                0x0000021952E4A150
		UWORD                       81 <SQL_GETDATA_EXTENSIONS>
		PTR                 0x000002195534DBF0
		SWORD                        4 
		SWORD *             0x0000007B2CFFDDDC

Microsoft.Mashu 2810-295c	EXIT  SQLGetInfoW  with return code 0 (SQL_SUCCESS)
		HDBC                0x0000021952E4A150
		UWORD                       81 <SQL_GETDATA_EXTENSIONS>
		PTR                 0x000002195534DBF0 ( 0x000000000000000F)
		SWORD                        4 
		SWORD *             0x0000007B2CFFDDDC (0)

Microsoft.Mashu 2810-295c	ENTER SQLNumResultCols 
		HSTMT               0x000002196D000530
		SWORD *             0x0000007B2CFFDF70

Microsoft.Mashu 2810-295c	EXIT  SQLNumResultCols  with return code 0 (SQL_SUCCESS)
		HSTMT               0x000002196D000530
		SWORD *             0x0000007B2CFFDF70 (0)

Microsoft.Mashu 2810-295c	ENTER SQLMoreResults 
		HSTMT               0x000002196D000530

Microsoft.Mashu 2810-295c	EXIT  SQLMoreResults  with return code 100 (SQL_NO_DATA_FOUND)
		HSTMT               0x000002196D000530

Microsoft.Mashu 2810-295c	ENTER SQLMoreResults 
		HSTMT               0x000002196D000530

Microsoft.Mashu 2810-295c	EXIT  SQLMoreResults  with return code 100 (SQL_NO_DATA_FOUND)
		HSTMT               0x000002196D000530

Microsoft.Mashu 2810-295c	ENTER SQLFreeHandle 
		SQLSMALLINT                  3 <SQL_HANDLE_STMT>
		SQLHANDLE           0x000002196D000530

Microsoft.Mashu 2810-295c	EXIT  SQLFreeHandle  with return code 0 (SQL_SUCCESS)
		SQLSMALLINT                  3 <SQL_HANDLE_STMT>
		SQLHANDLE           0x000002196D000530

Microsoft.Mashu 2810-295c	ENTER SQLDisconnect 
		HDBC                0x0000021952E4A150

Microsoft.Mashu 2810-295c	EXIT  SQLDisconnect  with return code 0 (SQL_SUCCESS)
		HDBC                0x0000021952E4A150

Microsoft.Mashu 2810-295c	ENTER SQLFreeHandle 
		SQLSMALLINT                  2 <SQL_HANDLE_DBC>
		SQLHANDLE           0x0000021952E4A150

Microsoft.Mashu 2810-295c	EXIT  SQLFreeHandle  with return code 0 (SQL_SUCCESS)
		SQLSMALLINT                  2 <SQL_HANDLE_DBC>
		SQLHANDLE           0x0000021952E4A150

Microsoft.Mashu 2810-295c	ENTER SQLAllocHandle 
		SQLSMALLINT                  2 <SQL_HANDLE_DBC>
		SQLHANDLE           0x0000021952E4A070
		SQLHANDLE *         0x00000219554467D0

Microsoft.Mashu 2810-295c	EXIT  SQLAllocHandle  with return code 0 (SQL_SUCCESS)
		SQLSMALLINT                  2 <SQL_HANDLE_DBC>
		SQLHANDLE           0x0000021952E4A070
		SQLHANDLE *         0x00000219554467D0 ( 0x0000021952E4A150)

Microsoft.Mashu 2810-295c	ENTER SQLSetConnectAttrW 
		SQLHDBC             0x0000021952E4A150
		SQLINTEGER                 103 <SQL_ATTR_LOGIN_TIMEOUT>
		SQLPOINTER                15
		SQLINTEGER                  -5 

Microsoft.Mashu 2810-295c	EXIT  SQLSetConnectAttrW  with return code 0 (SQL_SUCCESS)
		SQLHDBC             0x0000021952E4A150
		SQLINTEGER                 103 <SQL_ATTR_LOGIN_TIMEOUT>
		SQLPOINTER                15
		SQLINTEGER                  -5 

Microsoft.Mashu 2810-295c	ENTER SQLDriverConnectW 
		HDBC                0x0000021952E4A150
		HWND                0x0000000000000000
		WCHAR *             0x00007FFA1E8621A0 [      -3] "******\ 0"
		SWORD                       -3 
		WCHAR *             0x00007FFA1E8621A0 
		SWORD                       -3 
		SWORD *             0x0000000000000000
		UWORD                        0 <SQL_DRIVER_NOPROMPT>

Microsoft.Mashu 2810-295c	EXIT  SQLDriverConnectW  with return code 0 (SQL_SUCCESS)
		HDBC                0x0000021952E4A150
		HWND                0x0000000000000000
		WCHAR *             0x00007FFA1E8621A0 [      -3] "******\ 0"
		SWORD                       -3 
		WCHAR *             0x00007FFA1E8621A0 <Invalid buffer length!> [-3]
		SWORD                       -3 
		SWORD *             0x0000000000000000
		UWORD                        0 <SQL_DRIVER_NOPROMPT>

Microsoft.Mashu 2810-295c	ENTER SQLGetInfoW 
		HDBC                0x0000021952E4A150
		UWORD                        6 <SQL_DRIVER_NAME>
		PTR                 0x0000021955446BC0 
		SWORD                      100 
		SWORD *             0x0000007B2CFFDCEC

Microsoft.Mashu 2810-295c	EXIT  SQLGetInfoW  with return code 0 (SQL_SUCCESS)
		HDBC                0x0000021952E4A150
		UWORD                        6 <SQL_DRIVER_NAME>
		PTR                 0x0000021955446BC0 [      12] "DuckDB"
		SWORD                      100 
		SWORD *             0x0000007B2CFFDCEC (12)

Microsoft.Mashu 2810-295c	ENTER SQLGetInfoW 
		HDBC                0x0000021952E4A150
		UWORD                        7 <SQL_DRIVER_VER>
		PTR                 0x0000021955446E30 
		SWORD                      100 
		SWORD *             0x0000007B2CFFDCEC

Microsoft.Mashu 2810-295c	EXIT  SQLGetInfoW  with return code 0 (SQL_SUCCESS)
		HDBC                0x0000021952E4A150
		UWORD                        7 <SQL_DRIVER_VER>
		PTR                 0x0000021955446E30 [      20] "03.00.0000"
		SWORD                      100 
		SWORD *             0x0000007B2CFFDCEC (20)

Microsoft.Mashu 2810-295c	ENTER SQLGetInfoW 
		HDBC                0x0000021952E4A150
		UWORD                       17 <SQL_DBMS_NAME>
		PTR                 0x00000219554470C0 
		SWORD                      100 
		SWORD *             0x0000007B2CFFDCEC

Microsoft.Mashu 2810-295c	EXIT  SQLGetInfoW  with return code 0 (SQL_SUCCESS)
		HDBC                0x0000021952E4A150
		UWORD                       17 <SQL_DBMS_NAME>
		PTR                 0x00000219554470C0 [      12] "DuckDB"
		SWORD                      100 
		SWORD *             0x0000007B2CFFDCEC (12)

Microsoft.Mashu 2810-295c	ENTER SQLGetInfoW 
		HDBC                0x0000021952E4A150
		UWORD                       18 <SQL_DBMS_VER>
		PTR                 0x0000021955447330 
		SWORD                      100 
		SWORD *             0x0000007B2CFFDCEC

Microsoft.Mashu 2810-295c	EXIT  SQLGetInfoW  with return code 0 (SQL_SUCCESS)
		HDBC                0x0000021952E4A150
		UWORD                       18 <SQL_DBMS_VER>
		PTR                 0x0000021955447330 [      24] "v0.4.1-dev33"
		SWORD                      100 
		SWORD *             0x0000007B2CFFDCEC (24)

Microsoft.Mashu 2810-295c	ENTER SQLAllocHandle 
		SQLSMALLINT                  3 <SQL_HANDLE_STMT>
		SQLHANDLE           0x0000021952E4A150
		SQLHANDLE *         0x0000021955447BD0

Microsoft.Mashu 2810-295c	EXIT  SQLAllocHandle  with return code 0 (SQL_SUCCESS)
		SQLSMALLINT                  3 <SQL_HANDLE_STMT>
		SQLHANDLE           0x0000021952E4A150
		SQLHANDLE *         0x0000021955447BD0 ( 0x000002196D000530)

Microsoft.Mashu 2810-295c	ENTER SQLGetInfoW 
		HDBC                0x0000021952E4A150
		UWORD                       77 <SQL_DRIVER_ODBC_VER>
		PTR                 0x0000021955447C20 
		SWORD                      100 
		SWORD *             0x0000007B2CFFDE3C

Microsoft.Mashu 2810-295c	EXIT  SQLGetInfoW  with return code 0 (SQL_SUCCESS)
		HDBC                0x0000021952E4A150
		UWORD                       77 <SQL_DRIVER_ODBC_VER>
		PTR                 0x0000021955447C20 [      10] "03.00"
		SWORD                      100 
		SWORD *             0x0000007B2CFFDE3C (10)

Microsoft.Mashu 2810-295c	ENTER SQLGetInfoW 
		HDBC                0x0000021952E4A150
		UWORD                        6 <SQL_DRIVER_NAME>
		PTR                 0x0000021955448170 
		SWORD                      100 
		SWORD *             0x0000007B2CFFDE7C

Microsoft.Mashu 2810-295c	EXIT  SQLGetInfoW  with return code 0 (SQL_SUCCESS)
		HDBC                0x0000021952E4A150
		UWORD                        6 <SQL_DRIVER_NAME>
		PTR                 0x0000021955448170 [      12] "DuckDB"
		SWORD                      100 
		SWORD *             0x0000007B2CFFDE7C (12)

Microsoft.Mashu 2810-295c	ENTER SQLExecDirectW 
		HSTMT               0x000002196D000530
		WCHAR *             0x0000021955374104 [      -3] "load 'httpfs';\ a\ 0"
		SDWORD                    -3

Microsoft.Mashu 2810-295c	EXIT  SQLExecDirectW  with return code 0 (SQL_SUCCESS)
		HSTMT               0x000002196D000530
		WCHAR *             0x0000021955374104 [      -3] "load 'httpfs';\ a\ 0"
		SDWORD                    -3

Microsoft.Mashu 2810-295c	ENTER SQLGetFunctions 
		HDBC                0x0000021952E4A150
		UWORD                     1021 
		UWORD *             0x0000007B2CFFDF40

Microsoft.Mashu 2810-295c	EXIT  SQLGetFunctions  with return code 0 (SQL_SUCCESS)
		HDBC                0x0000021952E4A150
		UWORD                     1021 
		UWORD *             0x0000007B2CFFDF40 (1)

Microsoft.Mashu 2810-295c	ENTER SQLGetInfoW 
		HDBC                0x0000021952E4A150
		UWORD                       81 <SQL_GETDATA_EXTENSIONS>
		PTR                 0x0000021955448218
		SWORD                        4 
		SWORD *             0x0000007B2CFFDDDC

Microsoft.Mashu 2810-295c	EXIT  SQLGetInfoW  with return code 0 (SQL_SUCCESS)
		HDBC                0x0000021952E4A150
		UWORD                       81 <SQL_GETDATA_EXTENSIONS>
		PTR                 0x0000021955448218 ( 0x000000000000000F)
		SWORD                        4 
		SWORD *             0x0000007B2CFFDDDC (0)

Microsoft.Mashu 2810-295c	ENTER SQLNumResultCols 
		HSTMT               0x000002196D000530
		SWORD *             0x0000007B2CFFDF70

Microsoft.Mashu 2810-295c	EXIT  SQLNumResultCols  with return code 0 (SQL_SUCCESS)
		HSTMT               0x000002196D000530
		SWORD *             0x0000007B2CFFDF70 (0)

Microsoft.Mashu 2810-295c	ENTER SQLMoreResults 
		HSTMT               0x000002196D000530

Microsoft.Mashu 2810-295c	EXIT  SQLMoreResults  with return code 100 (SQL_NO_DATA_FOUND)
		HSTMT               0x000002196D000530

Microsoft.Mashu 2810-295c	ENTER SQLMoreResults 
		HSTMT               0x000002196D000530

Microsoft.Mashu 2810-295c	EXIT  SQLMoreResults  with return code 100 (SQL_NO_DATA_FOUND)
		HSTMT               0x000002196D000530

Microsoft.Mashu 2810-295c	ENTER SQLFreeHandle 
		SQLSMALLINT                  3 <SQL_HANDLE_STMT>
		SQLHANDLE           0x000002196D000530

Microsoft.Mashu 2810-295c	EXIT  SQLFreeHandle  with return code 0 (SQL_SUCCESS)
		SQLSMALLINT                  3 <SQL_HANDLE_STMT>
		SQLHANDLE           0x000002196D000530

Microsoft.Mashu 2810-295c	ENTER SQLDisconnect 
		HDBC                0x0000021952E4A150

Microsoft.Mashu 2810-295c	EXIT  SQLDisconnect  with return code 0 (SQL_SUCCESS)
		HDBC                0x0000021952E4A150

Microsoft.Mashu 2810-295c	ENTER SQLFreeHandle 
		SQLSMALLINT                  2 <SQL_HANDLE_DBC>
		SQLHANDLE           0x0000021952E4A150

Microsoft.Mashu 2810-295c	EXIT  SQLFreeHandle  with return code 0 (SQL_SUCCESS)
		SQLSMALLINT                  2 <SQL_HANDLE_DBC>
		SQLHANDLE           0x0000021952E4A150

**Reply by tiagokepe** [reply]
> it does work with windows CLI, and here is the log for ODBC with PowerQuery, notice that PowerQuery don't like query that does not return results

Hey @djouallah,
I made the same test with the PostgreSQL Odbc driver on PowerBI and on PGAdmin 4:

**With PowerBI the LOAD fails (my [ODBC trace file](https://github.com/duckdb/duckdb/files/9064390/ODBC.SQL.LOG) has the same log info that you posted here):**

![image](https://user-images.githubusercontent.com/403295/177789381-421da69a-8dcd-4ab4-8cb5-5d6463dceecb.png)

**With PGAdmin 4 it worked:**

![image](https://user-images.githubusercontent.com/403295/177789790-b84f2439-3c29-494b-a8b3-983289970e32.png)

My guess is the same like you had that PowerQuery _doesn't like query that does not return results_.

P.S. A workaround would be to add new key/word connection attribute into the connection string, which would be called `load_extensions` that receives a list of libraries to load, like:

`DSN=DuckDB; load_extensions=httpfs,parquet,json;`

During the connection we get that list and load all desired library specified by the `load_extensions` attribute.
What do you think @Mytherin ?
